flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/codegen/calls/ScalarOperators.scala

```diff
@@ -78,6 +78,69 @@ object ScalarOperators {
     }
   }
 
+  def generateIn(
+      nullCheck: Boolean,
+      left: GeneratedExpression,
+      right: scala.collection.mutable.Buffer[GeneratedExpression],
+      addReusableCodeCallback: (String, String) => Any)
+    : GeneratedExpression = {
+    val resultTerm = newName("result")
+    val isNull = newName("isNull")
+
+    val castNumeric = if (isNumeric(left.resultType) || isNumeric(right.head.resultType)) {
+      (value: String) => s"""new java.math.BigDecimal("" + $value).stripTrailingZeros()"""
```

Comment:
Casting all numeric values to `BigInteger` is quite expensive. Please check how numeric casting is done for binary operations, e.g., in `generateArithmeticOperator()`.